Fix part of #6240: Add audio translations for the state content

U8NWXD opened the pull request:
## Explanation

Fixes part of #6240 - end-to-end test to the exploration editor features to check that a user can add audio translations for the state content in the following ways (Previous PRs #8089, #8424):

- from recorded audio
- from file

The bulk of this PR was written by @piotrhm.

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python -m scripts.pre_commit_linter` and `python -m scripts.run_frontend_tests`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR has an appropriate "PR CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [x] The PR is **assigned** to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer and don't tick this checkbox.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue. Do not only request the review but also add the reviewer as an assignee.

piotrhm committed:
e2e test for adding audio translations

piotrhm committed:
Yarm update

piotrhm committed:
Fixing yarn

piotrhm committed:
Protractor config fixed

piotrhm committed:
merge

piotrhm committed:
test moved to voiceoverUploadFeatures.js

piotrhm committed:
Merge remote-tracking branch 'upstream/develop' into e2e-test-translations-audio

piotrhm committed:
fixed trailing spaces not allowed error

piotrhm committed:
fixed trailing spaces not allowed error

piotrhm committed:
Merge remote-tracking branch 'upstream/develop' into e2e-test-translations-audio

piotrhm committed:
Applied requested changes

piotrhm committed:
Applied requested changes 2

piotrhm committed:
Applied requested changes 3

piotrhm committed:
Fixed trailing issue

U8NWXD committed:
Address PR#8424 Review Comments

Makes the following changes:
* Specify audio file to use as fake audio source, and add comments to
  describe the needed Selenium arguments
* Remove duplicate `protractor-test-save-uploaded-audio` HTML class
* Condense audio deletion and confirming the deletion into a single
  function
* Clarify confusing variable names and error messages
* Fix how explorations are identified from the creator dashboard

Fixes part of #6240

U8NWXD committed:
Merge branch 'develop' into e2e-test-translations-audio

U8NWXD committed:
Merge branch 'develop' into e2e-test-translations-audio

U8NWXD committed:
Merge branch 'develop' of github.com:oppia/oppia into e2e-test-translations-audio

kevinlee12 commented on core/tests/protractor_desktop/voiceoverUploadFeatures.js:
```diff
@@ -85,6 +89,49 @@ describe('Voiceover upload features', function() {
+    explorationEditorTranslationTab.deleteAudioRecord();
+  });
+
+  it('should upload recorded file', function() {
```
Is it upload or successfully record a sound?

U8NWXD replied:
Yes, this name is confusing. Thanks for pointing this out! I'll fix it. This test uploads recorded audio, not a file chosen by the user.

U8NWXD replied:
Fixed by 6081d94bd

kevinlee12 reviewed (commented):
thanks, just a question!

U8NWXD committed:
Fix Confusing Audio Upload Test Name

The test for uploading recorded audio was named `should upload recorded
file`, which is confusing even though there might technically be an
audio file created by the browser from a recording and then uploaded. To
alleviate this confusion, the test is renamed to communicate that an
audio recording is being uploaded, not a file chosen by the user.

Fixes part of #6240. Addresses review comments on #8571 by @kevinlee12.

U8NWXD commented:
@kevinlee12 PTAL

kevinlee12 reviewed (approved):
lgtm for code owners

U8NWXD commented:
@nithusha21 @aks681 PTAL

nithusha21 commented on core/templates/dev/head/pages/exploration-editor-page/translation-tab/modal-templates/add-audio-translation-modal.template.html:
```diff
@@ -32,7 +32,7 @@ <h3>Add Audio Translation</h3>
 <div class="modal-footer">
   <button class="btn btn-secondary protractor-test-close-audio-upload-modal" ng-click="cancel()">Cancel</button>
-  <button class="btn btn-success protractor-test-save-uploaded-audio-button" ng-click="save()" ng-disabled="!isAudioTranslationValid() || saveInProgress">
+  <button class="btn btn-success  protractor-test-save-uploaded-audio" ng-click="save()" ng-disabled="!isAudioTranslationValid() || saveInProgress">
```
why these changes? I don't think you need the double space, and I think it is ok to suffix the class name with the type of object (button).

U8NWXD replied:
Resolved in f9c16e2d3

nithusha21 commented on core/tests/protractor_desktop/voiceoverUploadFeatures.js:
```diff
@@ -85,6 +89,49 @@ describe('Voiceover upload features', function() {
+    explorationEditorMainTab.exitTutorial();
+
+    explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.playAudioRecord();
```
This assumes the previous test ran before this? It might not always be the case I think (due to parallelization). Would be better to keep it separate from the above test.

U8NWXD replied:
Resolved by 70592c2e6

nithusha21 commented on core/tests/protractor_desktop/voiceoverUploadFeatures.js:
```diff
@@ -85,6 +89,49 @@ describe('Voiceover upload features', function() {
+    explorationEditorTranslationTab.playAudioRecord();
+  });
+
+  it('should play recorded file from file after logging out', function() {
```
Ditto here. I think for both of these, you can club the 2 it blocks and have just one test for both logged in and logged out flow.

U8NWXD replied:
Resolved by 70592c2e6

nithusha21 reviewed (commented):
LGTM overall. Just some small comments

aks681 commented on core/templates/dev/head/pages/exploration-editor-page/translation-tab/audio-translation-bar/audio-translation-bar.directive.html:
```diff
@@ -56,7 +56,7 @@
   </div>
   <div class="oppia-content-wrapper">
-    <button class="btn oppia-audio-button oppia-audio-bar-button-transition protractor-test-accessibility-translation-upload-audio protractor-test-upload-audio-button"
+    <button class="btn oppia-audio-button oppia-audio-bar-button-transition protractor-test-upload-audio protractor-test-accessibility-translation-upload-audio"
```
Why this change? We actually do suffix '-button' elsewhere in the codebase, so I'd suggest add it back here and add it to the other protractor classes that are added.

U8NWXD replied:
Fixed in 6993eb8c7

aks681 commented on core/templates/dev/head/pages/exploration-editor-page/translation-tab/audio-translation-bar/audio-translation-bar.directive.html:
```diff
@@ -92,7 +92,7 @@
   </div>
   <div class="oppia-content-wrapper" ng-if="voiceoverRecorder.status().isRecording">
-    <button class="btn oppia-audio-button oppia-audio-bar-button-transition"
+    <button class="btn oppia-audio-button oppia-audio-bar-button-transition protractor-test-stop-record"
```
Ditto here and elsewhere.

U8NWXD replied:
Fixed in 6993eb8c7

aks681 reviewed (approved):
Approving for now, just one comment on codeowner files about adding the -button suffix to the classes, else lgtm as codeowner.

aks681 commented:
Feel free to merge once that is addressed.

U8NWXD commented:
@kevinlee12 why did you merge this? I haven't addressed the review comments from @aks681 or @nithusha21 yet

kevinlee12 commented:
My apologizes, can you make a follow up PR that addresses these issues? thanks!

kevinlee12 commented:
I can also revert the PR if you'd like too.

U8NWXD commented:
@kevinlee12 I'll just make a follow-up PR since none of the requested fixes should affect correctness

kevinlee12 commented:
okay, thanks!

On Thu, Feb 13, 2020 at 7:21 PM U8N WXD <notifications@github.com> wrote:

> @kevinlee12 <https://github.com/kevinlee12> I'll just make a follow-up PR
> since none of the requested fixes should affect correctness
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/8571?email_source=notifications&email_token=ABL52MBLAYO3456BV354E4DRCYE2FA5CNFSM4KRIPXUKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELXNYTY#issuecomment-586079311>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABL52MA4TBO5AEPWTWFI6HDRCYE2FANCNFSM4KRIPXUA>
> .
>
